lms/services/d2l_api/client.py

```diff
@@ -44,3 +56,23 @@ def course_group_sets(self, org_unit):
         """
         response = self._api.request("GET", f"/{org_unit}/groupcategories/")
         return D2LGroupSetsSchema(response).parse()
+
+    def group_set_groups(self, org_unit, group_category_id, user_id=None):
+        """
+        Get the groups in a group category.
+
+        https://docs.valence.desire2learn.com/res/groups.html#post--d2l-api-lp-(version)-(orgUnitId)-groupcategories-(groupCategoryId)-groups-
+        """
+        response = self._api.request(
+            "GET",
+            f"/{org_unit}/groupcategories/{group_category_id}/groups/",
+        )
+        groups = D2LGroupsSchema(response).parse()
+        # D2L doesn't return the group_set_id of the listed groups as other LMS
+        # but we know which one it is because we queried for it, inject it:
+        groups = [dict(group, group_set_id=group_category_id) for group in groups]
```

Comment:
It's a little strange to redefine the object type here. I suspect they aren't different types though. Are we using a list comprehension to replace a loop here?

i.e. Could / should this be:

```suggestion
        for group in groups:
            group["group_set_id"] = group_category_id
```